### PR TITLE
Support InternVL2 model in VLM Pipeline

### PR DESCRIPTION
**Pull request by yatarkan**
Ticket: CVS-153334

### CODE REVIEW

File: .github/workflows/causal_lm_cpp.yml [hunk 1]
```diff
@@ -810,6 +810,17 @@ jobs:
           && ./build/samples/cpp/visual_language_chat/visual_language_chat ./llava_v1_6_mistral_7b_ov/ monalisa.jpg
           <<< $'Who drew this painting?\nWhen did the painter live?'
         timeout-minutes: 4
+      - name: Download and convert InternVL2 model and an image
+        run: |
+          source ./ov/setupvars.sh
+          optimum-cli export openvino --model OpenGVLab/InternVL2-4B ./internvl2_4b_ov/ --trust-remote-code
+          wget https://llava-vl.github.io/static/images/monalisa.jpg
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:817]: this picture is downloaded 2 times above.
Actually, we can keep only first download as next ones save file as monalisa.jpg.1, monalisa.jpg.2 etc

File: src/cpp/src/visual_language/input_embedder.cpp [hunk 2]
```diff
@@ -860,6 +860,108 @@ class InputsEmbedderLLaVANext : public InputsEmbedderLLaVA {
...
+        size_t image_context_tokens_count = 0;
+
+        for (size_t i = 0; i < flattened_size; ++i) {
+            if (input_ids_data[i] == image_context_token_id) {
+                image_context_tokens_mask[i] = true;
+                ++image_context_tokens_count;
+            }
+        }
+
+        OPENVINO_ASSERT(image_context_tokens_count > 0, "input_ids does not contain image context token ids");
```
Comment (ilya-lavrenov) [inline src/cpp/src/visual_language/input_embedder.cpp:944]: does it mean that we cannot perform generate w/o image? as `image_context_tokens_count` is required

Comment (yatarkan) [reply src/cpp/src/visual_language/input_embedder.cpp:944]: This part of code belongs to `merge_text_and_image_embeddings_internvl` method, that is executed only when images are provided and not empty.
So generate call w/o an image works

File: .github/workflows/causal_lm_cpp.yml (hunk 1 above)
Comment (yatarkan) [reply .github/workflows/causal_lm_cpp.yml]: Fixed

File: .github/workflows/causal_lm_cpp.yml [hunk 3]
```diff
@@ -799,17 +799,27 @@ jobs:
...
       - name: Run visual_language_chat C++ sample - LLaVa-Next
         run: >
           source ./ov/setupvars.sh
           && ./build/samples/cpp/visual_language_chat/visual_language_chat ./llava_v1_6_mistral_7b_ov/ monalisa.jpg
           <<< $'Who drew this painting?\nWhen did the painter live?'
         timeout-minutes: 4
+      - name: Download and convert InternVL2 model
+        run: |
+          source ./ov/setupvars.sh
+          python -m pip install timm
```
Comment (yatarkan) [inline .github/workflows/causal_lm_cpp.yml:815]: Probably, `timm` package is missing due to installing `"optimum<1.23"` with `--no-dependencies` flag above. AFAICS it is required for exporting InternVL2 model AFAICS.
Should be moved it to "Install dependencies" step above?

Comment (Wovchena) [reply .github/workflows/causal_lm_cpp.yml:815]: 1. `optimum` is installed with `python -m pip install --upgrade-strategy eager -r ./samples/requirements.txt opencv-python --pre --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly` so it should bring `timm`.
2. Later `transformers` from branch is installed.
3. There's compatibility issue and `optimum` is downgraded to solve this, but other packages aren't affected thanks to `--no-dependencies`.

If `timm` isn't installed, it should be added to samples and tests `requirements.txt`.

**Review by ilya-lavrenov — APPROVED** [review]

File: .github/workflows/causal_lm_cpp.yml (hunk 3 above)
Comment (yatarkan) [reply .github/workflows/causal_lm_cpp.yml]: Moved to samples requirements

**Review by Wovchena — DISMISSED** [review]

File: src/cpp/src/visual_language/input_embedder.cpp [hunk 4]
```diff
@@ -870,6 +972,8 @@ InputsEmbedder::InputsEmbedder(const VLMConfig& vlm_config,
         m_impl = std::make_shared<InputsEmbedderLLaVA>(vlm_config, model_dir, device, device_config);
     } else if (vlm_config.model_type == VLMModelType::LLAVA_NEXT) {
         m_impl = std::make_shared<InputsEmbedderLLaVANext>(vlm_config, model_dir, device, device_config);
+    } else if (vlm_config.model_type == VLMModelType::INTERNVL_CHAT) {
+        m_impl = std::make_shared<InputsEmbedderInternVL>(vlm_config, model_dir, device, device_config);
```
[the remark this answers is not in the document]
Comment (yatarkan) [reply src/cpp/src/visual_language/input_embedder.cpp]: Fixed

File: src/cpp/src/visual_language/input_embedder.cpp [hunk 5]
```diff
@@ -860,6 +860,108 @@ class InputsEmbedderLLaVANext : public InputsEmbedderLLaVA {
...
+            concated_image_tokens += image_start_token;
+            for (int i = 0; i < num_patches * num_image_tokens; ++i) {
+                concated_image_tokens += image_context_token;
+            }
+            concated_image_tokens += image_end_token;
+
+            size_t pos = formatted_prompt.find(image_token);
+            if (pos != std::string::npos) {
+                formatted_prompt.replace(pos, image_token.length(), concated_image_tokens);
+            }
```
[the remark this answers is not in the document]
Comment (yatarkan) [reply src/cpp/src/visual_language/input_embedder.cpp]: Fixed

**Review by Wovchena — APPROVED** [review]

**Pull request merged**